core/shared/src/main/scala/zio/ZLayer.scala

```diff
@@ -98,6 +98,12 @@ sealed abstract class ZLayer[-RIn, +E, +ROut] { self =>
   )(implicit tag: EnvironmentTag[ROut2]): ZLayer[RIn with RIn2, E1, ROut1 with ROut2] =
     self.++[E1, RIn2, ROut1, ROut2](that)
 
+  @deprecated("For macros only. Use ++ or and")
```

Comment:
I have a bad feeling that these annotations are going to cause issues with users getting deprecated warnings (maybe not with Scala 3.3 but with a future version).

Do we even need these methods? Can't we just change the macro code instead? Alternatively, what about adding them in a VersionSpecific trait, and use `private inline def` for them instead? That way they don't need to be public

Reply:
I actually thought I deleted them again. It was a thought, but I already had objections myself

Reply:
Yeah, having them inline would defeat the purpose. You could define these methods once in the generated code, although it's still pretty wasteful. Those shouldn't have been by-name in the first place (and they are eager in ZIO 1, not sure what happened).

Reply:
@987Nabil I just tried the changes in the PR and it seems that they're emitting deprecation warnings whenever the `ZLayer.make` macro is used:

<img width="718" alt="image" src="https://github.com/zio/zio/assets/67301607/38dbf9ea-9bb7-4c17-8694-31a4708da768">

Let's remove the deprecation annotation. Also, any chance making the method package-private works?